### PR TITLE
test: add invites endpoint tests

### LINKED ISSUE #66: [TEST] Unit Testing: invites and teams endpoint
# Test Task: `invites and teams data`

## Goal
Add tests for `invites.js` and `teams.js` to ensure correct behavior and edge case handling.

---

## Test Type
- [x] Unit (logic / handler)
- [ ] E2E (full request flow)

---

## Done When
- All tests pass in CI or locally if pipeline testing not setup yet
- Core flows + edge cases covered
- No failing lint/tests

### PR DESCRIPTION
**Pull request by bkj05**
Closes #66

### CODE REVIEW

**Comment by Amormio25** [comment]
@bkj05 
Important notes:
- it seems like you just removed a lot of existing code that was already working in the invites endpoint
- you put both tests (for invites and teams endpoints) in one file, when they should probably be two separate files as they're both pretty lengthy and need separate testing files even though some aspects may overlap (like the fact that we invite to a team)

Something that will help you is make some helper functions in both test files that creates some seed data locally that you can test on (e.g. create a helper function that initializes a team with mock members so you can test things like deleting users from teams or members leaving

Like I said if you have any trouble or confusion, it is better you ask immediately instead of plugging into chatgpt or something and it just removes a lot of existing functionality. 

Recall what we fixed in the invites and teams endpoints were listed here:
https://docs.google.com/document/d/1nkjU3QtpLkdKqh1ockLQgQD0Qsr5itWTWNbiU9HYTNA/edit?tab=t.1n77gd51c4ad#heading=h.ubksqd8jriyg
and if you check the status of code in main, you'll see the endpoints commented with what they do.

**Review by Amormio25 — CHANGES_REQUESTED** [review]
@bkj05 
Mostly good tests passed on my side, couple things:
- look at Jonathan's PR which had something like createTestSession, we need to make sure action is valid and authorized for each path like GET /invites, GET /invites/:id etc
- For PATCH and DELETE, you should also have 404 cases where the invite doesn't exist similar to how you have it in GET /invites/:id
- In PATCH /invites/:id/accept or reject, you need to do checks on id to make sure it's a valid number id which would be 400 error as well as for DELETE
- for POST do these checks:
  - has invited user id but no team id in payload
  - has team but no invited user
  - invalid team or user id
  - same for email

**Review by Amormio25 — CHANGES_REQUESTED** [review]
@bkj05  you currently have 2 PRs up right now, the one for the teams endpoint test, and the one for the invites endpoint test. 

But the invites endpoint test PR also has the a teams test file and it's different from the teams PR one.

The invite PR has a 786 lines of code file for the teams endpoint test, while the teams PR has a 531 lines of code file for the teams endpoint test.

Idk which one is the version you currently have, but I want you to fix it so that the invites PR only has the invites endpoint test file, and the teams PR only has the teams endpoint test whichever one is your final version

File: issue-tracker-api/test/invites.spec.js [hunk 1]
```diff
@@ -0,0 +1,644 @@
...
+
+				expect(response.status).toBe(200);
+				const data = await response.json();
+				expect(data.id).toBe(inviteId);
+				expect(data.team_name).toBe('Backend Team');
+				expect(data.inviter_username).toBe('alex');
+				expect(data.status).toBe('pending');
+			});
+
+			it('200: allows inviter to view the invite', async () => {
```
Comment (Amormio25) [inline issue-tracker-api/test/invites.spec.js]: inviter (admin) and invitee hould be able to view the invite, make sure inviter is admin

Refer to the updated invites endpoint it has some comments on this

**Review by Amormio25 — CHANGES_REQUESTED** [review]
@bkj05 just one small request, make sure the people who can view the invite is the admin and invitee

**Review by Amormio25 — APPROVED** [review]
